### DISCUSSION TITLE
Applying [up-switch] to radio buttons

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#339] [Q&A]
Is there a way to make [up-switch] to work with radio buttons?

Docs show only examples to select, checkboxes and text inputs. I tried to apply it to radio buttons but the compiler seems not to find the [up-switch] attribute on this element.

TIA

**Comment by adam12** [comment]
Can you make a reproduction using our [empty Glitch template](https://glitch.com/edit/#!/empty-unpoly-project)? I'm pretty sure this should work.

**Reply by alexandremjacques** [reply]
Unless I'm missing something...
https://glitch.com/edit/#!/gabby-boundless-rover?path=index.html%3A25%3A41

**Reply by adam12** [reply]
Looks like a bug. Removing wrapping `<form>` parent fixes it.

https://glitch.com/edit/#!/successful-nettle-muscari?path=index.html%3A23%3A19

**Reply by adam12** [reply]
Actually, maybe it's not a bug. If you have the `<input>` inside a `<form>`, Unpoly expects the targets to be inside that same `<form>`.

We'll have to see if Henning remembers if this is intentional. I think we should update the docs if so. And add a `radio` button example while we're at it.

**Reply by triskweline** [reply]
No strong opinion. I never wanted to switch something outside of the same form, but I'm open to changing this.
